src/common/list-syntax.k

```diff
@@ -0,0 +1,309 @@
+syntax SwitchGroupList ::= List{SwitchGroup, ""} [klabel('SwitchGroupList)]
+syntax SwitchLabelList ::= List{SwitchLabel, ""} [klabel('SwitchLabelList)]
+
+
+syntax UserList ::=     BlockStmList
+                      | SwitchGroupList
+                      | SwitchLabelList
+                      | InterfaceTypeList
+                      | ExceptionTypeList
+                      | IdList
+                      | TypeDecList
+                      | VarDecList
+                      | ImportDecList
+                      | ActualTypeArgList
+                      | ClassOrInterfaceTypeList
+                      | TypeParamList
+                      | AbstractMethodModList
+                      | AnnoAbstractMethodModList
+                      | AnnoMethodModList
+                      | AnnoVarModList
+                      | AnnoClassModList
+                      | AnnoConstrModList
+                      | AnnoConstantModList
+                      | AnnoFieldModList
+                      | AnnoInterfaceModList
+                      | AnnoList
+                      | AnnoElemDecList
+                      | InterfaceMemberDecList
+                      | ElemValPairList
+                      | ElemValList
+                      | StringPartList
+                      | EnumConstList
+                      | ClassBodyDecList
+
+/* Syntax below should be put in other modules when they are perfectly defined. */
+                      syntax   InterfaceType
+                      syntax   ExceptionType
+                      syntax   Id
+                      syntax   TypeDec::= ClassDec
+                      syntax   VarDec
+                      syntax   ImportDec
+                      syntax   ActualTypeArg
+                      syntax   ClassOrInterfaceType
+                      syntax   TypeParam
+                      syntax   AbstractMethodMod
+                      syntax   AnnoAbstractMethodMod
+                      syntax   AnnoMethodMod
+                      syntax   AnnoVarMod
+                      syntax   AnnoClassMod
+                      syntax   AnnoConstrMod
+                      syntax   AnnoConstantMod
+                      syntax   AnnoFieldMod
+                      syntax   AnnoInterfaceMod
+                      syntax   Anno
+                      syntax   AnnoElemDec
+                      syntax   InterfaceMemberDec
+                      syntax   ElemValPair
+                      syntax   ElemVal
+                      syntax   StringPart
+                      syntax   EnumConst
+                      syntax   ClassBodyDec
+
+endmodule
+
+module LIST-MACRO
+imports CORE-FUNCTIONS
+
+syntax KItem ::= toKListWrap(K) [function]
+                | toKListWrap(K, KListWrap) [function]
+
+rule toKListWrap(UL) => toKListWrap(UL,[.KList])
+
+rule toKListWrap(Label:KLabel(Hd,,Tl), [L:KList]) => toKListWrap(Tl, [L,,Hd])
+
+rule toKListWrap('.List`{""`}(.KList), KL:KListWrap) => KL
```

Comment:
Oh yes, clever, actually the (.KList) is enough to judge if it is the terminator, your rule also applies to other seperators if they do appear. Good idea!